### PR TITLE
Add Arm intrinsics to JpegColorConverter RGB

### PR DESCRIPTION
**Pull request by stefannikolei**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
This is my first attempt in learning something about intrinsics. 
So I picked something which looked straight forward (for me) for a port to arm.

I will also try to see if there is already an benchmark for this, or going to look into writing one

### CODE REVIEW

**Comment by brianpopow** [comment]
> I will also try to see if there is already an benchmark for this, or going to look into writing one

There is a benchmark comparing the AVX version to the RgbVector version in Codecs/Jpeg/ColorConverion/RgbColorConversion.cs

You can make a similar one for the ARM version.

**Comment by stefannikolei** [comment]
|        Method |      Mean |    Error |   StdDev | Ratio |
|-------------- |----------:|---------:|---------:|------:|
|        Scalar | 323.78 ns | 0.979 ns | 0.916 ns |  1.00 |
|   SimdVector8 | 110.25 ns | 1.073 ns | 0.951 ns |  0.34 |
| SimdVectorArm |  90.27 ns | 1.841 ns | 2.581 ns |  0.28 |

**Comment by brianpopow** [comment]
> Method	Mean	Error	StdDev	Ratio
> Scalar	323.78 ns	0.979 ns	0.916 ns	1.00
> SimdVector8	110.25 ns	1.073 ns	0.951 ns	0.34
> SimdVectorArm	90.27 ns	1.841 ns	2.581 ns	0.28

nice, I wasn't sure at first that this will beat the `RgbVector` version, but this benchmark looks good!

**Comment by JimBobSquarePants** [comment]
Our `UnscopedRefAttribute` is not working on the ARM CI server. 

It looks like they're using a different version of MSBuild

MSBuild version 17.3.2+561848881 for .NET

Verses GitHub's 17.5.0-preview-23061-01+040e2a90e

**Comment by stefannikolei** [comment]
> Our `UnscopedRefAttribute` is not working on the ARM CI server. 
> 
> 
> 
> It looks like they're using a different version of MSBuild
> 
> 
> 
> MSBuild version 17.3.2+561848881 for .NET
> 
> 
> 
> Verses GitHub's 17.5.0-preview-23061-01+040e2a90e
> 
> 

When I look at it, they do it right. 17.3 is the expected MSBuild version for net6 (https://learn.microsoft.com/de-de/dotnet/core/porting/versioning-sdk-msbuild-vs)
It will probably work when we define this build task like this:

Sdk: 7.0.x
Framework: 6.0.x

**Comment by JimBobSquarePants** [comment]
Yeah, changing the DotNet Setup task to use 7 might work.

**Comment by JimBobSquarePants** [comment]
I think we're hitting issues in the tests because the VMs are underpowered. We can bump them up by changing the signature.

Let's try `buildjet-8vcpu-ubuntu-2204-arm` first and if that doesn't help, bump it up again.
https://buildjet.com/for-github-actions/docs/runners#arm

**Comment by stefannikolei** [comment]
> I think we're hitting issues in the tests because the VMs are underpowered. We can bump them up by changing the signature.
> 
> Let's try `buildjet-8vcpu-ubuntu-2204-arm` first and if that doesn't help, bump it up again. https://buildjet.com/for-github-actions/docs/runners#arm

I removed the .net6 target for arm. But now hitting another issue god dammit...

File: src/ImageSharp/Formats/Jpeg/Components/ColorConverters/JpegColorConverter.RgbArm.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            ref Vector128<float> rBase =
+                ref Unsafe.As<float, Vector128<float>>(ref MemoryMarshal.GetReference(values.Component0));
+            ref Vector128<float> gBase =
+                ref Unsafe.As<float, Vector128<float>>(ref MemoryMarshal.GetReference(values.Component1));
+            ref Vector128<float> bBase =
+                ref Unsafe.As<float, Vector128<float>>(ref MemoryMarshal.GetReference(values.Component2));
+
+            // Used for the color conversion
+            var scale = Vector128.Create(1 / this.MaximumValue);
+            nint n = values.Component0.Length / Vector128<float>.Count;
```
[the remark this answers is not in the document]
Comment (JimBobSquarePants) [reply src/ImageSharp/Formats/Jpeg/Components/ColorConverters/JpegColorConverter.RgbArm.cs:33]: Will have to bear that in mind. We've likely got similar repeated through the codebase

[the remark this answers is not in the document]
Comment (gfoidl) [reply src/ImageSharp/Formats/Jpeg/Components/ColorConverters/JpegColorConverter.RgbArm.cs]: I'll do a pass over the codebase and fix them up.

[the remark this answers is not in the document]
Comment (JimBobSquarePants) [reply src/ImageSharp/Formats/Jpeg/Components/ColorConverters/JpegColorConverter.RgbArm.cs]: That'd be ace. Thanks!

**Review by JimBobSquarePants — APPROVED** [review]
This is great. Thanks!

**Comment by brianpopow** [comment]
I think we missed here adjusting the [JpegColorConverterTests](https://github.com/SixLabors/ImageSharp/blob/5283d7736f3546c67b63d8a95d26830bb83f4836/tests/ImageSharp.Tests/Formats/Jpg/JpegColorConverterTests.cs#L23 ) for ARM. It should now also run the tests with `DisableArm64AdvSimd`, right?

**Comment by stefannikolei** [comment]
> I think we missed here adjusting the [JpegColorConverterTests](https://github.com/SixLabors/ImageSharp/blob/5283d7736f3546c67b63d8a95d26830bb83f4836/tests/ImageSharp.Tests/Formats/Jpg/JpegColorConverterTests.cs#L23 ) for ARM. It should now also run the tests with `DisableArm64AdvSimd`, right?

I can add that in the new pr I made. But what does that do? It will then run some tests instead of 2 times 3 times and disables specific simds

**Comment by brianpopow** [comment]
> > I think we missed here adjusting the [JpegColorConverterTests](https://github.com/SixLabors/ImageSharp/blob/5283d7736f3546c67b63d8a95d26830bb83f4836/tests/ImageSharp.Tests/Formats/Jpg/JpegColorConverterTests.cs#L23) for ARM. It should now also run the tests with `DisableArm64AdvSimd`, right?
> 
> I can add that in the new pr I made. But what does that do? It will then run some tests instead of 2 times 3 times and disables specific simds

This will run the tests using `FeatureTestRunner`, which as you noticed let you disable hardware intrinsics features. The point of this is to test the Scalar path and the intrinsics path. Now on the ARM CI (assuming the CI supports intrinsics), only the intrinsics path is tested.

I think the following IntrinsicsConfig should now be used:
```
private const HwIntrinsics IntrinsicsConfig = HwIntrinsics.AllowAll | HwIntrinsics.DisableHWIntrinsic;
```

On x64, we will execute the scalar version with `HwIntrinsics.DisableHWIntrinsic` and execute the AVX version with `HwIntrinsics.AllowAll`.
On ARM, we will execute the scalar version with `HwIntrinsics.DisableHWIntrinsic` and execute the hardware intrinsics version with `HwIntrinsics.AllowAll`.

**Comment by JimBobSquarePants** [comment]
Do we not have various granular implementations though? Avx, Sse etc? We want to turn them off incrementally not en-mass.

**Comment by brianpopow** [comment]
> Do we not have various granular implementations though? Avx, Sse etc? We want to turn them off incrementally not en-mass.

That's in general true, but we only have AVX version for x64 CPU's, no SSE variants for the color conversion.

The alternative would be: `HwIntrinsics.AllowAll | HwIntrinsics.DisableAVX | HwIntrinsics.DisableArm64AdvSimd` (right?)

This will do the following on x64, assuming AVX is available for the CPU:
- `HwIntrinsics.AllowAll`: Run AVX version
- `HwIntrinsics.DisableAVX`: Run Scalar version
- `HwIntrinsics.DisableArm64AdvSimd`: Again run AVX version (correct?)

On ARM:
- `HwIntrinsics.AllowAll`: Run ARM version
- `HwIntrinsics.DisableAVX`: Run ARM version
- `HwIntrinsics.DisableArm64AdvSimd`: Run Scalar version

Thats why I suggested 'HwIntrinsics.AllowAll | HwIntrinsics.DisableHWIntrinsic'

**Pull request merged**